lib/Doctrine/ORM/Query/SqlWalker.php

```diff
@@ -883,9 +883,39 @@ public function walkIdentificationVariableDeclaration($identificationVariableDec
      */
     public function walkIndexBy($indexBy)
     {
-        $pathExpression = $indexBy->simpleStateFieldPathExpression;
+        $pathExpression = $indexBy->singleValuedPathExpression;
         $alias          = $pathExpression->identificationVariable;
-        $field          = $pathExpression->field;
+
+        /** @var Query\AST\PathExpression $pathExpr */
+        switch ($pathExpression->type) {
+            case AST\PathExpression::TYPE_STATE_FIELD:
+                $field = $pathExpression->field;
+                break;
+
+            case AST\PathExpression::TYPE_SINGLE_VALUED_ASSOCIATION:
+                // 1- the owning side:
+                //    Just use the foreign key, i.e. u.group_id
+                $fieldName   = $pathExpression->field;
+                $class       = $this->queryComponents[$alias]['metadata'];
+                $association = $class->getProperty($fieldName);
+
+                if (! $association->isOwningSide()) {
+                    throw QueryException::associationPathInverseSideNotSupported($pathExpr);
+                }
+
+                $joinColumns = $association->getJoinColumns();
+
+                // COMPOSITE KEYS NOT (YET?) SUPPORTED
```

Comment:
they can't be supported for this, so its ok to remove the comment. The exception is exactly showing users this is not supported.

Reply:
The switch here is copied from walkPathExpression(), including the comments. Happy to remove the comment if you like though.

Reply:
Yes i guess removing the comment makes sense then, also at the other location.